.github/workflows/pr-tests.yaml

```
@@ -0,0 +1,55 @@
+
+jobs:
+  backend-style:
+    runs-on: ubuntu-latest
+    name: Run Backend Pre-Commit
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
```

actionlint (1.7.4)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)